server/src/Repositories/genericRepository.js

```diff
@@ -0,0 +1,59 @@
+import mongoose from 'mongoose';
+
+class Repository {
+  constructor(model) {
+    this.Model = model;
+  }
+
+  static generateObjectID() {
+    String(mongoose.Types.ObjectId());
+  }
+
+  create(data) {
+    return this.Model.create({ ...data, _id: this.generateObjectID() });
+  }
+
+  countUser() {
```

Comment:
this can be just `count` instead of `countUser`

Reply:
Ohh, typo. I'll change it.